otherlibs/systhreads/st_stubs.c

```diff
@@ -488,6 +494,10 @@ CAMLprim value caml_thread_initialize(value unit)
   return Val_unit;
 }
 
+/* Cleanup the thread machinery when the runtime is shut down. Joining the tick
+   thread take 25ms on average / 50ms in the worst case, so we don't do it on
+   program exit. (FIXME: not implemented in OCaml 5 yet) */
```

Comment:
IIUC, the fix is to not join the systhread since it might be sleeping?

Reply:
Note, as I mentioned in https://github.com/ocaml/ocaml/pull/11386#issuecomment-1456384769, that not joining is more problematic for multicore, and I suggested the use of pthread_cond_timedwait with a condition variable used to request continuing or shutting down as a possible solution.